tests/use_cases/classify/test_classify.py

```diff
@@ -240,3 +270,74 @@ def test_multi_label_classify_evaluator_full_dataset(
     assert {"positive", "negative", "finance", "school"} == set(
         aggregation_overview.statistics.class_metrics.keys()
     )
+
+
+def test_single_label_classify_with_file_repository(
+    tmp_path: Path,
+) -> None:
+    in_memory_dataset_repository = FileDatasetRepository(tmp_path)
+    examples = [
+        Example(
+            input=ClassifyInput(
+                chunk=TextChunk("I am happy."),
+                labels=frozenset(["happy", "sad", "angry"]),
+            ),
+            expected_output="happy",
+        ),
+        Example(
+            input=ClassifyInput(
+                chunk=TextChunk("I am sad."),
+                labels=frozenset(["happy", "sad", "angry"]),
+            ),
+            expected_output="sad",
+        ),
+        Example(
+            input=ClassifyInput(
+                chunk=TextChunk("I am angry."),
+                labels=frozenset(["happy", "sad", "angry"]),
+            ),
+            expected_output="angry",
+        ),
+    ]
+    dataset_id = in_memory_dataset_repository.create_dataset(
+        examples=examples, dataset_name="single-label-classify"
+    ).id
+
+    in_memory_run_repository = FileRunRepository(tmp_path)
+    classify_runner = Runner(
+        PromptBasedClassify(),
+        in_memory_dataset_repository,
+        in_memory_run_repository,
+        "single-label-classify",
+    )
+    run_overview = classify_runner.run_dataset(dataset_id)
```

Comment:
it would be really nice to not use a llm task here and instead just create a dummy classifier that creates the format that you need. It might also be overkill to start from a dataset when you want to test only the aggregation

Reply:
The tests are already slow as it is and if we can not use llamas wherever possible, it makes for faster and less flaky pipelines